### PR TITLE
build sysroot flag

### LINKED ISSUE #478: Rework the option of the build system
Currently, the way of building rustc_codegen_gcc without building the sysroot is:
```
./y.sh test --build-only
```

It would be better if `./y.sh build` would do this, so `--build-only` will not be an option for the command `test` anymore.

I see two options here:

 * Add a `--sysroot` flag to `./y.sh build` that will build the sysroot only when this flag is specified.
 * ~Do `./y.sh build --build-only` to not build the sysroot.~

### PR DESCRIPTION
**Pull request by lordshashank**
- resolves https://github.com/rust-lang/rustc_codegen_gcc/issues/478
- `--sysroot` flag added to build sysroot only when its true

### CODE REVIEW

**Comment by lordshashank** [comment]
@antoyo only m68k CI is failing, should we add 
`        CG_GCC_TEST_TARGET=m68k-unknown-linux-gnu cargo build -Zbuild-std`
in m68k build CI?
There is no breaking change here, idk why it fails.

**Comment by antoyo** [comment]
You seem to be missing the `--sysroot` flag [here](https://github.com/antoyo/rustc_codegen_gcc/blob/master/.github/workflows/m68k.yml#L88).

Perhaps your branch is not up-to-date with the master branch?

**Comment by lordshashank** [comment]
@antoyo let me know if anything else is needed here

File: build_system/src/build.rs [hunk 1]
```diff
@@ -31,6 +32,9 @@ impl BuildArg {
                         );
                     }
                 }
+                "--sysroot" => {
+                    build_arg.build_sysroot = true; // Set the flag when --sysroot is encountered
```
Comment (antoyo) [inline build_system/src/build.rs]: Please remove the comment as I find it redundant with the code.

File: build_system/src/build.rs [hunk 2]
```diff
@@ -205,9 +210,11 @@ fn build_codegen(args: &mut BuildArg) -> Result<(), String> {
     let _ = fs::remove_dir_all("target/out");
     let gccjit_target = "target/out/gccjit";
     create_dir(gccjit_target)?;
-
-    println!("[BUILD] sysroot");
-    build_sysroot(&env, &args.config_info)?;
+    if args.build_sysroot {
+        // Only call build_sysroot if the flag is set
```
Comment (antoyo) [inline build_system/src/build.rs]: Please remove the comment as I find it redundant with the code.

**Comment by lordshashank** [comment]
@antoyo, seems to be network error.

**Comment by antoyo** [comment]
Yeah, I restarted the job.

**Comment by lordshashank** [comment]
LFG 🚀

**Comment by antoyo** [comment]
Thanks for your contribution!

**Review by antoyo — APPROVED** [review]

**Pull request merged**